### PR TITLE
[fix][pass] Save the function when it is used as a call arg

### PR DESCRIPTION
**Pull request by zhiics**
Currently the args of the call nodes are not checked when removing functions. It causes function undefined problem when a global function is invoked by a call as its argument because the definition will be removed.

cc @wweic @icemelon9

### CODE REVIEW

File: src/relay/backend/vm/removed_unused_funcs.cc [hunk 1]
```diff
@@ -76,6 +72,15 @@ struct CallTracer : ExprVisitor {
...
+    } else {
+      VisitExpr(expr);
+    }
+  }
+
+  void VisitExpr_(const CallNode* call_node) final {
+    CheckExpr(call_node->op);
+    for (auto param : call_node->args) {
+      CheckExpr(param);
     }
```
Comment (wweic) [inline src/relay/backend/vm/removed_unused_funcs.cc]: Can the bug be fixed by adding the code here? Looks like the bug is we don't continue visit the expression if it's not FunctionNode or GlobalVarNode. 
````python
else {
      VisitExpr(expr);
    }
````

Comment (zhiics) [reply src/relay/backend/vm/removed_unused_funcs.cc]: Yes, it could be. I just moved it out to make it look more clear. How do you think?

Comment (wweic) [reply src/relay/backend/vm/removed_unused_funcs.cc]: Works for me.

**Comment by icemelon9** [comment]
Thanks @zhiics @wweic

**Pull request merged**